src/block-range/domain/use-cases/process-received-block.use-case.ts

```diff
@@ -0,0 +1,78 @@
+import { BlockRangeScanRepository } from '@common/block-range-scan/domain/repositories/block-range-scan.repository';
+import { BlockRange } from '@common/block-range/domain/entities/block-range';
+import { GetBlocksResult } from '@common/state-history/domain/entities/get-blocks.result';
+import { log } from '@common/state-history/domain/state-history.utils';
+import { inject, injectable } from 'inversify';
+import { Result } from '@core/architecture/domain/result';
+import { UseCase } from '@core/architecture/domain/use-case';
+import { QueueActionProcessingUseCase } from '@common/data-processing-queue/domain/use-cases/queue-action-processing.use-case';
+
+/**
+ * @class
+ */
+@injectable()
+export class ProcessReceivedBlockUseCase implements UseCase<void> {
+  public static Token = 'PROCESS_RECEIVED_BLOCK_USE_CASE';
+
+  @inject(QueueActionProcessingUseCase.Token)
+  private queueActionProcessingUseCase: QueueActionProcessingUseCase;
+  @inject(BlockRangeScanRepository.Token)
+  private blockRangeScanRepository: BlockRangeScanRepository;
+
+  /**
+   * Initializes and returns a eosDAC StateReceiver object.
+   *
+   * @param {Block} block
+   * @returns {Result<void>}
+   */
+  public async execute(
+    scanKey: string,
+    content: GetBlocksResult,
+    blockRange: BlockRange
+  ): Promise<Result<void>> {
```

Comment:
Specifying return type is a code noise? Its better to know what function returns when you start reading the code block not only from reading its body or using editor plugins for TS. We could also drop specifying  "public" as it is default in TS but code must be readable for everyone

Reply:
Just like getSomeString1 function. I think less type description means less to read. Or I just saw to much F# :)

type Result<T> = { key: T }

const getSomeString1 = (test: string) => {
  var someResult = { key: test } as Result<string>
  return someResult;
}

const getSomeString2 = (test: string): Result<string> => {
  var someResult: Result<string> = { key: test } as Result<string>
  return someResult;
}

Reply:
Typing is like description / code instruction. If you start skipping part of this description, it will make the code less readable, understandable. Of course, in the code itself, if it is explicit and readable, you don't need to type variables (even lint is discouraged if it concerns basic types).

But in declaring functions, class methods, using types helps you instantly see what you need and get. Do not keep looking at the fact that the transpiler is smart enough and he just gets what is going on. You write code for other developers (juniors, middle guys or aces and it's supposed to be consistent and readable).

lets drop typed params :)

Reply:
Another example, let's say you have

```
getMeSomethigNice(foo, bar) {
  // some magic
  return addThisAndDoThatAndReturn(foo, bar, ...);
}
```
In this function you have some operations but additionally you call a different method to make further modifications and return your "something". Without using return type you have to search the code to guess what you will get at the end by calling `getMeSomethingNice`

Typing is not essential but it makes life easier and helps keep things clean. Without typing "coders" will flood the code with their various "creative inventions" and then in other places add `if's` (their result or other)